src/IdProviderFactory.php

```diff
@@ -59,13 +59,20 @@ private static function dbExecute() {
 	}
 
 	/**
-	 * Checks whether a WikiPage with the following id/title already exists
+	 * Function to check if string id/title already exists, i.e. is already used as a
```

Comment:
Documentation:

The PHPDoc is clear but could mention that the method returns a closure and explain why this design choice was made. For example: